src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallGlobalOrToolPathCommand.cs

```diff
@@ -17,7 +19,8 @@
 namespace Microsoft.DotNet.Tools.Tool.Install
 {
     internal delegate IShellShimRepository CreateShellShimRepository(string appHostSourceDirectory, DirectoryPath? nonGlobalLocation = null);
-    internal delegate (IToolPackageStore, IToolPackageStoreQuery, IToolPackageInstaller) CreateToolPackageStoresAndInstaller(
+    
+    internal delegate (IToolPackageStore, IToolPackageStoreQuery, IToolPackageDownloader) CreateToolPackageStoresAndDownloader(
```

Comment:
Can you also delete the `IToolPackageInstaller` interface and everything that uses it?  I think by now anything that uses it should be dead code.

Reply:
Sure. I wonder if it's a good idea to submit another PR for deleting `IToolPackageInstaller` related code. I'm worried if the change is involved in too many files that it will affect other code

Reply:
I think we should try doing it in this PR.  There may be a lot of code to delete, but I think all of it is now dead code (ie won't be used), and code deletions like that are easy to review.